contracts/mocks/Nft.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: CC0-1.0
+
+pragma solidity 0.8.9;
+
+import { ERC721 } from '@openzeppelin/contracts/token/ERC721/ERC721.sol';
+import { Base64 } from '@openzeppelin/contracts/utils/Base64.sol';
+
+contract Nft is ERC721 {
+  string rolesMetadata;
+
+  constructor() ERC721('Nft', 'NFT') {}
+
+  function mint(address to, uint256 tokenId) external {
+    _mint(to, tokenId);
+  }
+
+  function tokenURI(uint256 tokenId) public view override returns (string memory) {
+    bytes memory dataURI = abi.encodePacked(
+      '{',
+      '"name": "Nft name",',
+      '"description": "Nft description",',
+      '"image": "Nft image",',
+      '"roles": [',
+      rolesMetadata,
+      ']',
+      '}'
+    );
+    return string(abi.encodePacked('data:application/json;base64,', Base64.encode(dataURI)));
+  }
+
+  function setRolesMetadata(string memory _rolesMetadata) external {
+    rolesMetadata = _rolesMetadata;
+  }
```

Comment:
Here we are storing the metadata of the NFTs on-chain, which is not wrong. However, to represent most of the NFT cases that we will have, we should just return an URL here with the tokenId in the path.

We should also use a mock server and an http client on tests to represent the return url.